AndroidSDK/src/com/leanplum/Leanplum.java

```diff
@@ -706,176 +712,184 @@ private static void handleApiResponse(JSONObject response, List<Map<String, Obje
 
     if (hasStartResponse) {
       if (!LeanplumInternal.hasStarted()) {
+        // Set start response to null.
+        req.onApiResponse(null);
         Leanplum.handleStartResponse(lastStartResponse);
       }
     }
   }
 
-  private static void handleStartResponse(JSONObject response) {
-    boolean success = Request.isResponseSuccess(response);
-    if (!success) {
-      try {
-        LeanplumInternal.setHasStarted(true);
-        LeanplumInternal.setStartSuccessful(false);
+  private static void handleStartResponse(final JSONObject response) {
+    Util.executeAsyncTask(false, new AsyncTask<Void, Void, Void>() {
```

Comment:
we need to handle Start response in the different thread, at this time synchronized block will work correctly, when we will call getNewsfeedMessages.